aten/src/ATen/core/ivalue.h

```diff
@@ -365,11 +365,14 @@ struct TORCH_API IValue final {
       return payload.as_tensor.use_count();
     }
 
-    if (!is_intrusive_ptr) {
+    if (!isIntrusivePtr()) {
       return 1;
     }
 
     if (payload.u.as_intrusive_ptr == c10::UndefinedTensorImpl::singleton()) {
+      if (isStorage() || isGenerator()) {
+        return 1;
+      }
```

Comment:
This deserves a comment.

Reply:
changing to use isIntrusivePtrLegacyBehavior